cpp/include/cudf/io/types.hpp

```diff
@@ -63,6 +70,7 @@ enum class io_type {
   HOST_BUFFER,               ///< Input/output is a buffer in host memory,
   ARROW_RANDOM_ACCESS_FILE,  ///< Input/output is an arrow::io::RandomAccessFile
   VOID,                      ///< Input/output is nothing. No work is done. Useful for benchmarking
+  USER_SINK,                 ///< Input/output is handled by a custom user class
```

Comment:
While you are at it, suggest you give this enum class an explicit datatype (`enum class io_type : int32_t {`) like we are for the rest of libcudf.